### PR TITLE
add inner for jax.numpy frontend

### LINKED ISSUE #9756: inner

### PR DESCRIPTION
**Pull request by YanniPapandreou**
Close #9756

### CODE REVIEW

**Comment by YanniPapandreou** [comment]
I think I have implemented the `jax.numpy.inner` function but I am having issues with the associated frontend test. At first I was getting the `empty suite` error in pycharm referenced in the docs. The solution suggested there did not work for me but I managed to fix it by specifying the working directory as the `ivy/ivy_tests` folder. Now I am running into the following error for the tests which I am unsure of how to fix:

```
test setup failed
file /ivy/ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_linalg.py, line 852
  def test_jax_numpy_inner(
E       fixture 'dtype_and_x' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, doctest_namespace, frontend, include_metadata_in_junit_xml, json_metadata, metadata, monkeypatch, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, run_around_tests, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/ivy/ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_linalg.py:852
```

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by Nightcrab** [comment]
The function has been added to the wrong file. ``jax.numpy.inner`` is in the ``jax.numpy`` namespace, not ``jax.numpy.linalg``. The definition of inner should be in ``jax/numpy/mathematical_functions.py``.

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_linalg.py [hunk 1]
```diff
@@ -836,3 +836,39 @@ def test_jax_numpy_cond(
         a=x[0],
         p=p,
     )
+
+
+# inner
+handle_frontend_test(
```
[the remark this answers is not in the document]
Comment (YanniPapandreou) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_linalg.py]: I fixed this but I am still getting a similar error about fixtures not found. I am not sure what is wrong.

**Comment by YanniPapandreou** [comment]
Thanks @Nightcrab I have moved the files to correct location. I am still getting errors on the test run which I am having trouble debugging.

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py [hunk 2]
```diff
@@ -2163,3 +2163,39 @@ def test_jax_numpy_real(
...
+        min_value=-10,
+        max_value=10,
+        num_arrays=2,
+        shared_dtype=True,
+    ),
+)
+def test_jax_numpy_inner(
+    *,
+    dtype_and_x,
+    as_variable,
```
Comment (Nightcrab) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py:2182]: Remove this

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py [hunk 3]
```diff
@@ -2163,3 +2163,39 @@ def test_jax_numpy_real(
...
+    num_positional_args,
```
Comment (Nightcrab) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py:2183]: This too

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py [hunk 4]
```diff
@@ -2163,3 +2163,39 @@ def test_jax_numpy_real(
...
+    native_array,
```
Comment (Nightcrab) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py:2184]: Replace this with ``test_flags``

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py [hunk 5]
```diff
@@ -2163,3 +2163,39 @@ def test_jax_numpy_real(
...
+    on_device,
+    fn_tree,
+    frontend,
+):
+    input_dtypes, xs = dtype_and_x,
+    helpers.test_frontend_function(
+        input_dtypes=input_dtypes,
+        as_variable_flags=as_variable,
+        with_out=False,
```
Comment (Nightcrab) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py:2193]: Remove ``with_out`` and the arguments that you removed above. Add ``test_flags=test_flags,``.

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py [hunk 6]
```diff
@@ -2163,3 +2163,39 @@ def test_jax_numpy_real(
...
+    *,
+    dtype_and_x,
+    as_variable,
+    num_positional_args,
+    native_array,
+    on_device,
+    fn_tree,
+    frontend,
+):
+    input_dtypes, xs = dtype_and_x,
```
Comment (Nightcrab) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py:2189]: Remove the comma after ``dtype_and_x``

**Review by Nightcrab — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py (hunk 2 above)
Comment (YanniPapandreou) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py]: Removed this1

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py (hunk 3 above)
Comment (YanniPapandreou) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py]: And this too.

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py (hunk 4 above)
Comment (YanniPapandreou) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py]: I've changed this to `test_flags`

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py (hunk 6 above)
Comment (YanniPapandreou) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py]: Ahh thanks for catching that, I've fixed this typo :)

**Comment by YanniPapandreou** [comment]
The test runs now and passes, it just throws the following warning: 
```
2023-02-07 09:30:57.173580: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcuda.so.1'; dlerror: libcuda.so.1: cannot open shared object file: No such file or directory
2023-02-07 09:30:57.173642: W tensorflow/stream_executor/cuda/cuda_driver.cc:269] failed call to cuInit: UNKNOWN ERROR (303)
```

I'm guessing this might be because I do not have a gpu on my machine.

Is there anything else I need to do for this frontend task? The linting seems to be failing so I'll try to fix that.

**Comment by Nightcrab** [comment]
All looks fine to me, merging.

**Comment by YanniPapandreou** [comment]
Thanks @Nightcrab for your help :)
